src/inmanta/deploy/scheduler.py

```diff
@@ -108,6 +108,28 @@ class ResourceIntentChange(Enum):
     """
 
 
+class _ResourceIntentChange(abc.ABC):
+    change: ClassVar[ResourceIntentChangeType]
+
+
+class New(_ResourceIntentChange):
+    change = ResourceIntentChangeType.NEW
+
+
+class Updated(_ResourceIntentChange):
+    change = ResourceIntentChangeType.UPDATED
+
+
+@dataclass(frozen=True, kw_only=True)
+class Deleted(_ResourceIntentChange):
+    change: ClassVar[ResourceIntentChangeType] = ResourceIntentChangeType.DELETED
```

Comment:
I'm curious why `: ClassVar[ResourceIntentChangeType]` here and not for the other 2 classes?

Reply:
Here because `dataclass` requires it. For the others I didn't think the verbosity was worth it. But I'm not opposed to adding it there.